### ISSUE TITLE
R2 values questionable

### ISSUE THREAD

**Issue by robertcrowe** [tflearn/tflearn#469]
It looks like there is the same problem with tflearn.metrics.WeightedR2 that we saw earlier with tflearn.metrics.R2:

> Exception: Weighted R2 metric requires Inputs and Targets to have same shape.

And I'm also getting R2 values > 1, which don't make sense.  I'm beginning to wonder about the R2 code in general.

**Comment by robertcrowe** [comment]
I have a suggested revision, does this look right to you?
```
y_pred = tf.to_double(y_pred)
y_true = tf.to_double(y_true)

numerator = tf.reduce_sum(tf.squared_difference(y_true, y_pred))
denominator = tf.reduce_sum(tf.squared_difference(y_true, tf.reduce_mean(y_true)))
zero = tf.zeros(tf.shape(denominator), dtype=tf.float64)
one = tf.ones(tf.shape(denominator), dtype=tf.float64)
denominator = tf.cond(tf.equal(denominator, zero), lambda: one, lambda: denominator) # avoid NaN

return tf.truediv(numerator, denominator)
```
